woocommerce/woocommerce-android
# Issue/4044 Basic detekt integration

Closes Issue #4044

Following [the post] here is the first step of Detekt integration in our project

paqN3M-jK-p2

It contains:
* Baseline (without formatting)
* Config file
* Detekt plugin connection
* Detekt-formatter plugin connection

Please go through my comments in the  PR

### How to test
Run `./gradlew detektAll` and notice some code formatting violations


@woocommerce/platform-9 someone please take a look as well, just in case
